Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -483,6 +262,45 @@ void AnalyzeLepTrigger::WriteHistos(TFile* outfile)
     
 }
 
+bool AnalyzeLepTrigger::containsGoodLepton( const std::vector<TLorentzVector>& leptons, const std::vector<bool>& goodLeptons, double ptThreshold, double etaSelection) { 
+    //Require a good muon in the single muon dataset
+    for( unsigned int iLep = 0; iLep < leptons.size(); ++iLep ) {
+        if( !goodLeptons.at( iLep ) ) continue; 
+    
+        TLorentzVector myLepton = leptons.at( iLep );
+    
+        if( myLepton.Pt() >= ptThreshold && std::fabs( myLepton.Eta() ) < etaSelection ) return true;
+    }
+
+    return false;
+}
+
+int AnalyzeLepTrigger::goodLeptonIndex( const std::vector<TLorentzVector>& leptons, const std::vector<bool>& goodLeptons) {
+    for( unsigned int iLep = 0; iLep < leptons.size(); ++iLep ) {
+        if( !goodLeptons.at( iLep ) ) continue;
+
+        return iLep;
+    }
+
+    return -1;
+}
+
+void AnalyzeLepTrigger::fillHistos( const std::map<std::string, bool>& cutMap, bool passLeptonTriggers, const TLorentzVector& lepton, double theWeight ) {
+    for( auto& kv : cutMap ) {
+        if( kv.second ) {
+            my_histos["h_trig_den_"+kv.first+"_wLepPtBin"]->Fill( lepton.Pt(), theWeight );
+            my_histos["h_trig_den_"+kv.first+"_wLepEtaBin"]->Fill( lepton.Eta(), theWeight );
+            my_2d_histos["h2_trig_den_"+kv.first+"_wLepPtLepEtaBin"]->Fill( lepton.Pt(), lepton.Eta(), theWeight );
+
+            if( passLeptonTriggers ) {
+                my_histos["h_trig_num_"+kv.first+"_wLepPtBin"]->Fill( lepton.Pt(), theWeight );
+                my_histos["h_trig_num_"+kv.first+"_wLepEtaBin"]->Fill( lepton.Eta(), theWeight );
+                my_2d_histos["h2_trig_num_"+kv.first+"_wLepPtLepEtaBin"]->Fill( lepton.Pt(), lepton.Eta(), theWeight );
+            }
+        }
+    }
+}
+
 //This is just the general trigger passing function
 bool AnalyzeLepTrigger::passTriggerGeneral( std::vector<std::string>& myTriggerVector, const std::vector<std::string>& TriggerNames, const std::vector<int>& TriggerPass ) {
```

Comment:
This function can be removed since you don't check if the trigger was passed in this file.

Reply:
Done